app/models/pieces/pawn.rb

```diff
@@ -1,2 +1,30 @@
 class Pawn < Piece
+  # rubocop:disable Metrics/LineLength 
+  def valid_move?(row_dest, col_dest)
+  	row_diff = (row_position - row_dest).abs
+    col_diff = (col_position - col_dest).abs
+  	case 
+  	when self.backward_move?(row_dest, col_dest) then false
+  	when !self.obstructed?(row_dest, col_dest) && self.diagonal_move?(row_dest, col_dest) &&  col_diff == 1 && row_diff == 1 && game.pieces.find_by(row_position: row_dest, col_position: col_dest) && !self.own_piece?(row_dest, col_dest) then true
+    when !self.obstructed?(row_dest, col_dest) && self.vertical_move?(row_dest, col_dest) && !game.pieces.find_by(row_position: row_dest, col_position: col_dest) then
+  	  if row_position == 1 || row_position == 6 
+  	    if row_diff <= 2
+  	    	return true
+  	    elsif row_diff == 1
+  	    	return true
+  	    else
+  	    	return false
+  	    end	 
+  	  end 
+  	  else
+  	  return false 		
+  	end
```

Comment:
I like how you used case statements, but I find it kind of difficult to read sometimes with the long `when`s. Is there a particular reason you chose to use case statements here? :smile: 
I'm also thinking there is a more elegant way to execute all of those checks. 
For example, we could check if the piece is obstructed before the whole case statements to prevent the redundant `.obstructed?` call.
What do you think?